### PR TITLE
Add wasm publish workflow

### LINKED ISSUE #1435: Add wasm release workflow

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Added wasm publish workflow from https://github.com/iotaledger/iota.rs-wasm/blob/develop/.github/workflows/release-wasm-bindings-to-npm.yml

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota.rs/issues/1435

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/release-wasm-bindings-to-npm.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Wasm publish to NPM
+
+on:
+  workflow_dispatch:
```
Comment (thibault-martinez) [inline .github/workflows/release-wasm-bindings-to-npm.yml]:
```suggestion
on: workflow_dispatch
```

**Review by thibault-martinez — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]
LGTM, but we could make it less error prone by setting a variable for the directory

https://github.com/orgs/community/discussions/25742#discussioncomment-3249014

File: .github/workflows/release-wasm-bindings-to-npm.yml [hunk 2]
```diff
@@ -32,19 +34,19 @@ jobs:
 
       - name: Install Yarn
         run: npm i -g yarn
-        working-directory: client/bindings/wasm
+        working-directory: ${{ env.working-directory }}
```
Comment (thibault-martinez) [inline .github/workflows/release-wasm-bindings-to-npm.yml:37]: Just checking, is the underscore automatically transformed into a dash ?

Comment (Thoralf-M) [reply .github/workflows/release-wasm-bindings-to-npm.yml:37]: Probably not, messed up

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**